### DISCUSSION TITLE
Multiple names for single Function object

### DISCUSSION THREAD

**Discussion by SyxP** [Vector35/binaryninja-api#4849] [API] [answered]
Is it possible to assign multiple names to a single function object (at the same address). The usecase is where I have a compiled code, and I know that multiple functions (due to compilation) are compiled to the same function address, but occasionally, I would still want to be able to search either name in a function panel and still recover the function.

**Comment by psifertex** [comment, marked as answer]
I is possible to create multiple symbols using the API, however right now the UI will only show one FunctionSymbol (the most recent one) so you won't be able to search for all of them.

That said, there _is_ a small hack which is that you can create DataVariable symbols at the same location as the function and show data symbols and you'll be able to see all of them in the UI at the same time. Consider this for example:

1. First, turn on data symbols:
<img width="1026" alt="Screenshot 2024-01-03 at 14 47 15" src="https://github.com/Vector35/binaryninja-api/assets/140215/6b3dd8d3-bf64-40a6-b90b-4c10a3da90eb">

2. Next, add a new datasymbol:
```python
bv.define_user_symbol(Symbol(SymbolType.DataSymbol, here, "NewFunctionName"))
```

3. This will "rename" the function, but the old name will still show up on the symbol list along with the new name:
<img width="1859" alt="Screenshot 2024-01-03 at 14 48 25" src="https://github.com/Vector35/binaryninja-api/assets/140215/079834ec-2566-43db-b30b-3122674b2859">

**Reply by psifertex** [reply]
Also, see https://github.com/Vector35/binaryninja-api/issues/4902 for an issue tracking improvements to the Symbols pane to duplicate entries.
